src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs

```diff
@@ -7,131 +7,271 @@ public interface IS3Repository : IRepository<IS3RepositorySettings> { }
 	{
 		internal S3RepositorySettings() { }
 
-		public S3RepositorySettings(string bucket)
-		{
-			this.Bucket = bucket;
-		}
+		public S3RepositorySettings(string bucket) => this.Bucket = bucket;
 
+		/// <inheritdoc />
 		public string Bucket { get; set; }
-		[Obsolete("Removed in NEST 6.x.")]
+		/// <inheritdoc />
 		public string Region { get; set; }
+		/// <inheritdoc />
+		public string Endpoint { get; set; }
+		/// <inheritdoc />
+		public string Protocol { get; set; }
+		/// <inheritdoc />
 		public string BasePath { get; set; }
+		/// <inheritdoc />
 		public string AccessKey { get; set; }
+		/// <inheritdoc />
 		public string SecretKey { get; set; }
+		/// <inheritdoc />
+		public string ChunkSize { get; set; }
+		/// <inheritdoc />
 		public bool? Compress { get; set; }
+		/// <inheritdoc />
 		public int? ConcurrentStreams { get; set; }
-		public string ChunkSize { get; set; }
+		/// <inheritdoc />
+		public bool? ServerSideEncryption { get; set; }
+		/// <inheritdoc />
+		public string BufferSize { get; set; }
+		/// <inheritdoc />
+		public int? MaximumRetries { get; set; }
+		/// <inheritdoc />
+		public bool? UseThrottleRetries { get; set; }
+		/// <inheritdoc />
+		public bool? ReadOnly { get; set; }
+		/// <inheritdoc />
+		public string CannedAcl { get; set; }
+		/// <inheritdoc />
+		public string StorageClass { get; set; }
+		/// <inheritdoc />
+		public bool? PathStyleAccess { get; set; }
 	}
 
 	public class S3RepositorySettingsDescriptor
 		: DescriptorBase<S3RepositorySettingsDescriptor, IS3RepositorySettings>, IS3RepositorySettings
 	{
 		string IS3RepositorySettings.Bucket { get; set; }
-		[Obsolete("Removed in NEST 6.x.")]
 		string IS3RepositorySettings.Region { get; set; }
+		string IS3RepositorySettings.Endpoint { get; set; }
+		string IS3RepositorySettings.Protocol { get; set; }
 		string IS3RepositorySettings.BasePath { get; set; }
 		string IS3RepositorySettings.AccessKey { get; set; }
 		string IS3RepositorySettings.SecretKey { get; set; }
-		bool? IS3RepositorySettings.Compress { get; set; }
-		int? IS3RepositorySettings.ConcurrentStreams { get; set; }
 		string IS3RepositorySettings.ChunkSize { get; set; }
+		int? IS3RepositorySettings.ConcurrentStreams { get; set; }
+		bool? IS3RepositorySettings.Compress { get; set; }
+		bool? IS3RepositorySettings.ServerSideEncryption { get; set; }
+		string IS3RepositorySettings.BufferSize { get; set; }
+		int? IS3RepositorySettings.MaximumRetries { get; set; }
+		bool? IS3RepositorySettings.UseThrottleRetries { get; set; }
+		bool? IS3RepositorySettings.ReadOnly { get; set; }
+		string IS3RepositorySettings.CannedAcl { get; set; }
+		string IS3RepositorySettings.StorageClass { get; set; }
+		bool? IS3RepositorySettings.PathStyleAccess { get; set; }
 
-		/// <summary>
-		/// The name of the bucket to be used for snapshots. (Mandatory)
-		/// </summary>
-		/// <param name="bucket"></param>
+		public S3RepositorySettingsDescriptor(string bucket) => Self.Bucket = bucket;
```

Comment:
Also need a public parameterless ctor to maintain BWC

Reply:
:+1: